apps/query-server/queryserver/main.py

```diff
@@ -67,29 +97,158 @@ async def get_index(
+    return plan
 
 
-@app.post("/v1/query")
-async def run_query(query: Query) -> QueryResult:
-    """Generate a plan for the given query, run it, and return the result."""
+def doc_to_json(doc: Document) -> Optional[dict[str, Any]]:
+    """Render a Document as a JSON object. Returns None for MetadataDocuments."""
+    NUM_TEXT_CHARS_GENERATE = 1024
+
+    if isinstance(doc, MetadataDocument):
+        return None
+
+    props_dict = {}
+    props_dict.update(doc.properties)
+    if "_schema" in props_dict:
+        del props_dict["_schema"]
+    if "_schema_class" in props_dict:
+        del props_dict["_schema_class"]
+    if "_doc_source" in props_dict:
+        del props_dict["_doc_source"]
+    props_dict["text_representation"] = (
+        doc.text_representation[:NUM_TEXT_CHARS_GENERATE] if doc.text_representation is not None else None
+    )
+    return props_dict
+
+
+async def run_query_stream(query: Query) -> EventSourceResponse:
+    """Streaming version of run_query. Returns a stream of results as they are generated."""
+
+    async def query_runner():
+        try:
+            logger.info(f"Generating plan for {query.index}: {query.query}")
+            yield {
+                "event": "status",
+                "data": "Generating plan",
+            }
+            await asyncio.sleep(0.1)
+            plan = sqclient.generate_plan(query.query, query.index, sqclient.get_opensearch_schema(query.index))
+            logger.info(f"Generated plan: {plan}")
+            # Don't want to return these through the API.
+            plan.llm_plan = None
+            plan.llm_prompt = None
+            yield {
+                "event": "plan",
+                "data": plan.model_dump_json(),
+            }
+            await asyncio.sleep(0.1)
+            logger.info("Running plan")
+            yield {
+                "event": "status",
+                "data": "Running plan",
+            }
+            await asyncio.sleep(0.1)
+            sqresult = sqclient.run_plan(plan)
+            t1 = time.time()
+            num_results = 0
+            if isinstance(sqresult.result, DocSet):
+                logger.info("Got DocSet result")
+                for doc in sqresult.result.take_all():
+                    rendered = doc_to_json(doc)
+                    logger.info(f"Doc: {rendered}")
+                    if rendered is not None:
+                        num_results += 1
+                        yield {
+                            "event": "result_doc",
+                            "data": rendered,
+                        }
+                        await asyncio.sleep(0.1)
+            else:
+                num_results += 1
+                yield {
+                    "event": "result",
+                    "data": sqresult.result,
+                }
+                await asyncio.sleep(0.1)
+
+            for doc in sqresult.retrieved_docs():
```

Comment:
This turns out to not work. If the node has a summarize, then the results can get truncated and retrieved_docs as written does the wrong thing by returning partial results since the code forces materialize to run.  A fixed version that looks for materialize.success also fails because it finds no directory with complete results.
To get it to work, I stripped the summarize step off the end of the plan in the code in the customer environment since right now we only care about the set of documents.
Also for the customer this doesn't happen to return what they want, path is the wrong property to return.

Reply:
This does work for me, but even so, your comment is irrelevant to this PR. If we need a fix to `retrieved_docs` we can do that separately.